bin/templates/scripts/cordova/lib/prepare.js

```diff
@@ -272,37 +272,46 @@ function updateBuildPropertyLocal (proj, targetName, prop, value, build) {
     try {
         // Check if we have a valid target - during prepare we do not have it
         const target = proj.pbxTargetByName(targetName);
-        if (!target || !target.buildConfigurationList) {
-            proj.updateBuildProperty(prop, value, build);
-        } else {
-            // Collect the uuid's from the configuration of our target
-            const COMMENT_KEY = /_comment$/;
-            const validConfigs = [];
-            const configList = proj.pbxXCConfigurationList();
-            for (const configName in configList) {
-                if (!COMMENT_KEY.test(configName) && target.buildConfigurationList === configName) {
-                    const buildVariants = configList[configName].buildConfigurations;
-                    for (let i = 0; i < buildVariants.length; i++) {
-                        validConfigs.push(buildVariants[i].value);
-                    }
-                    break;
+        const targetBuildConfigs = target && target.buildConfigurationList;
+
+        // Go to fallback in the catch and update build properties
+        if (!targetBuildConfigs) throw new CordovaError(`The target "${targetName}" is missing build configurations. Falling back to update build properties.`);
```

Comment:
```suggestion
        if (!targetBuildConfigs) {
           throw new CordovaError(`The target "${targetName}" is missing build configurations. Falling back to update build properties.`);
        }
```

Reply:
I am not a fan of these one line ifs but that is personal preference.

Reply:
Using the catch makes sense to avoid duplicating the fallback 👍 

But it is harder to understand for me while reading the code, but I would merge this change.

Reply:
Typically I like to keep things on one line when possible. Specifically for throwing errors or return.. But if it becomes too long then I would break it out like this..

I know others had also agreed with what I said and brought up the same argument. But others as well bring up your point..

I will let you decide since it is your PR.. I am OK with either..

Reply:
I fine with both too so I will leave it as is.